vassal-app/src/main/java/VASSAL/counters/Deck.java

```diff
@@ -882,17 +882,22 @@ public Command shuffle() {
     return setContents(l).append(reportCommand(shuffleMsgFormat, Resources.getString("Deck.shuffle"))); //$NON-NLS-1$
   }
 
-  /** Shuffle the contents of the Deck, IF it is an always-shuffle, and we're about to ask for a random list from it */
-  public Command maybeShuffle() {
+  /**
+   * Return a list if pieces in the Deck in Dealable order.
+   * If this is an Always shuffle Deck, then shuffle the list of pieces, otherwise just
+   * reverse the list order so that we deal from the top.
+   *
+   * @return List of pieces in Dealable order
+   */
+  public List<GamePiece> getOrderedPieces() {
+    final List<GamePiece> pieces = asList();
     if (!ALWAYS.equals(shuffleOption)) {
-      return new NullCommand();
+      Collections.shuffle(pieces);
```

Comment:
Should that be `Collections.shuffle(pieces, gameModule.getRNG());`, to ensure we're using the right RNG?

Reply:
Indeed.

Reply:
Am I misreading this, or is this code only shuffling if the shuffleOption is NOT "Always"?  Isn't that backwards?

Reply:
The dangers of cut & paste. You are 100% correct. Good pick-up :)